sliceop/f64s/f64s.go

```diff
@@ -62,3 +66,42 @@ func Find(dst []int, src []float64, f func(v float64) bool) []int {
 
 	return dst
 }
+
+// Take creates a sub-slice of src with all elements indiced by the provided indices.
+// Take uses dst as work buffer, storing elements at the start of the slice.
+// Take clears dst if a slice is passed, and allocates a new slice if dst is nil.
+// Take will panic if indices is not sorted or has duplicates.
+// Take will panic if length of indices is larger than length of src.
+// Take will panic if length of indices is different from length of dst.
+func Take(dst, src []float64, indices []int) []float64 {
+
+	if !sort.IntsAreSorted(indices) {
+		panic("f64s: indices not sorted")
+	}
+
+	if len(indices) > len(src) {
+		panic("f64s: length mismatch")
+	}
+
+	if dst == nil {
+		dst = make([]float64, len(indices))
+	}
+
+	if len(dst) != len(indices) {
+		panic("f64s: length mismatch")
+	}
+
+	dst = dst[0:len(indices)]
+
+	m := make(map[int]int, len(indices))
+	for i, v := range indices {
+		if _, there := m[v]; !there {
```

Comment:
I am not sure it wouldn't be faster to do this in 2 loops:
- one to fill the "set" of indices (we tend to use a `map[T]struct{}` for this: `struct{}` does not take up any memory),
- then to fill `dst`.

this can be part of a later PR (I'll play a bit with benchmarks)

Reply:
OK sounds good to me! I have replaced the `int` by en empty structure in the meanwhile.